services/pull/pull.go

```diff
@@ -311,23 +311,39 @@ func checkIfPRContentChanged(pr *models.PullRequest, oldCommitID, newCommitID st
 func PushToBaseRepo(pr *models.PullRequest) (err error) {
 	log.Trace("PushToBaseRepo[%d]: pushing commits to base repo '%s'", pr.BaseRepoID, pr.GetGitRefName())
 
+	// Clone base repo.
+	tmpBasePath, err := models.CreateTemporaryPath("pull")
```

Comment:
if func `PushToBaseRepo` is called in paralel - they still result in different tmpRepos ?
@zeripath

Reply:
Yup they do. Take a look at the code for create temporary path.